### PR TITLE
fix(sdk): correct JSDoc and enhance code readability for api on `tf-aws` platform

### LINKED ISSUE #3919: Unable to use path variables in cloud.Api on tf-aws
### I tried this:

```js
bring cloud;
let api = new cloud.Api();
api.get("/{foo}", inflight (req) => { });
```

### This happened:

```
ERROR: Invalid path /{foo}. Url cannot contain ":", params contains only alpha-numeric chars or "_".
```

### I expected this:

To work.

### Is there a workaround?

_No response_

### Component

SDK

### Wing Version

_No response_

### Node.js Version

_No response_

### Platform(s)

_No response_

### Anything else?

_No response_

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### LINKED ISSUE #5210: api path: collision with /{proxy+}
### I tried this:

A minimal example:

```
bring cloud;

let api = new cloud.Api();

api.get("/:id", inflight (req) => {
    let id = req.vars.get("id");

    return {
        body: "id: {id}",
        status: 200,
    };
});
```

### This happened:

It compiles, but fails when I try to apply the tf files:

```
Error: creating API Gateway REST API (hjnq3g5xqd) specification: BadRequestException: Errors found during import:
│       Unable to create resource at path '/{proxy+}': A sibling ({id}) of this resource already has a variable path part -- only one is allowed
│ Additionally, these warnings were found:
│       Parse issue: attribute info is missing
│       Required "info" property is missing from the document root.
│ 
│   with aws_api_gateway_rest_api.cloudApi_api_2B334D75,
│   on main.tf.json line 72, in resource.aws_api_gateway_rest_api.cloudApi_api_2B334D75:
│   72:       }
```

### I expected this:

API route in `/:id`.

### Is there a workaround?

_No response_

### Anything else?

Here is the JSON the program generated:

```json
{
    "openapi": "3.0.3",
    "paths": {
        "/{id}": {
            "get": {
                "operationId": "get-:id",
                "responses": {
                    "200": {
                        "description": "200 response",
                        "content": {}
                    }
                },
                "parameters": [
                    {
                        "name": "id",
                        "in": "path",
                        "required": true,
                        "schema": {
                            "type": "string"
                        }
                    }
                ],
                "x-amazon-apigateway-integration": {
                    "uri": "arn:aws:apigateway:${data.aws_region.Region.name}:lambda:path/2015-03-31/functions/${aws_lambda_function.cloudApi_get_id_0_456E8BD7.arn}/invocations",
                    "type": "aws_proxy",
                    "httpMethod": "POST",
                    "responses": {
                        "default": {
                            "statusCode": "200"
                        }
                    },
                    "passthroughBehavior": "when_no_match",
                    "contentHandling": "CONVERT_TO_TEXT"
                }
            }
        },
        "/{proxy+}": {
            "x-amazon-apigateway-any-method": {
                "produces": [
                    "application/json"
                ],
                "x-amazon-apigateway-integration": {
                    "type": "mock",
                    "requestTemplates": {
                        "application/json": "\n {\"statusCode\": 404}\n "
                    },
                    "passthroughBehavior": "never",
                    "responses": {
                        "404": {
                            "statusCode": "404",
                            "responseParameters": {
                                "method.response.header.Content-Type": "'application/json'"
                            },
                            "responseTemplates": {
                                "application/json": "{\"statusCode\": 404, \"message\": \"Error: Resource not found\"}"
                            }
                        },
                        "default": {
                            "statusCode": "404",
                            "responseParameters": {
                                "method.response.header.Content-Type": "'application/json'"
                            },
                            "responseTemplates": {
                                "application/json": "{\"statusCode\": 404, \"message\": \"Error: Resource not found\"}"
                            }
                        }
                    }
                },
                "responses": {
                    "404": {
                        "description": "404 response",
                        "headers": {
                            "Content-Type": {
                                "type": "string"
                            }
                        }
                    }
                }
            }
        }
    }
}
```

The compiler always adds the "/{proxy+}" part and this causes a conflict with the variable that the user chooses.

I think the problem is in this line:

https://github.com/winglang/wing/blob/693ee4b3a995a9c305479b32c10e0a87013ff125/libs/wingsdk/src/target-tf-aws/api.ts#L325

It should be:

```typescript
const defaultResponse = props.cors ? API_CORS_DEFAULT_RESPONSE(props.cors) : {};
```

### Wing Version

0.51.16

### Node.js Version

v21.2.0

### Platform(s)

MacOS

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### PR DESCRIPTION
**Pull request by garysassano**
## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [ ] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

**Review by tsuf239 — COMMENTED** [review]
The 404 wing test is failing, can you please fix?

**Comment by staycoolcall911** [comment]
Hey @tsuf239, @garysassano  - do you know the status of this PR? Thanks!

**Comment by garysassano** [comment]
@staycoolcall911 It was actually ready to ship, but now it requires quite some changes since #5659 got merged and the complexity got even bigger; i personally find the code hard to follow.

It was actually ready to ship, but with the integration of #5659, the complexity has further increased, necessitating quite some changes. I find the code even harder to understand now.

**Comment by garysassano** [comment]
@staycoolcall911 It was actually ready to ship, but with the integration of https://github.com/winglang/wing/pull/5659, the complexity has further increased, necessitating quite some changes. I find the code even harder to understand now.

**Comment by garysassano** [comment]
It was actually ready to ship, but with the integration of https://github.com/winglang/wing/pull/5659, the complexity has further increased, necessitating quite some changes. I find the code even harder to read now.

**Comment by hasanaburayyan** [comment]
> It was actually ready to ship, but with the integration of #5659, the complexity has further increased, necessitating quite some changes. I find the code even harder to read now.

😁 sorry about that, yea 5659 was required to allow the `tf-aws` platform to deploy a private API gateway. Using platform parameters that were introduced in: https://github.com/winglang/wing/pull/5557

**Comment by hasanaburayyan** [comment]
Also nit on the title - this would be `tf-aws.Api` not `cloud.Api`

File: libs/wingsdk/src/target-tf-aws/api.ts [hunk 1]
```diff
@@ -304,18 +304,19 @@ export class Api extends cloud.Api implements IAwsApi {
...
   public readonly stage: ApiGatewayStage;
   public readonly deployment: ApiGatewayDeployment;
-  public readonly securityGroup?: SecurityGroup;
-  public readonly privateVpc: boolean;
-  public readonly vpcEndpoint?: VpcEndpoint;
-  private readonly region: string;
-  private readonly accountId: string;
+  public readonly privateVpc: boolean = false;
+  public securityGroup?: SecurityGroup;
+  public vpcEndpoint?: VpcEndpoint;
```
Comment (tsuf239) [inline libs/wingsdk/src/target-tf-aws/api.ts:319]: is there a way to keep those readonly?

**Review by tsuf239 — APPROVED** [review]
